Tests/InkTests/HTMLTests.swift

```diff
@@ -103,6 +103,121 @@ final class HTMLTests: XCTestCase {
 
         XCTAssertEqual(html, "<p>Hello</p><!-- Comment --><p>World</p>")
     }
+    
+    func testMarkdownHeadingInsideHTML() {
+        let html = MarkdownParser().html(from: """
+        <div>
+
+        ## Heading
+
+        </div>
+        """)
+
+        XCTAssertEqual(html, "<div><h2>Heading</h2></div>")
+    }
+    
+    func testMarkdownImageInsideHTML() {
+        let html = MarkdownParser().html(from: """
+        <div>
+
+        ![test image](https://test.com/test.jpg)
+
+        </div>
+        """)
+
+        XCTAssertEqual(html, "<div><img src=\"https://test.com/test.jpg\" alt=\"test image\"/></div>")
+    }
+    
+    func testMarkdownListInsideHTML() {
+        let html = MarkdownParser().html(from: """
+        <div>
+
+        - One
+        - Two
+        - Three
+
+        </div>
+        """)
+
+        XCTAssertEqual(html, "<div><ul><li>One</li><li>Two</li><li>Three</li></ul></div>")
+    }
+    
+    func testMarkdownBeforeHTML() {
+        let html = MarkdownParser().html(from: """
+        <div>
+
+        # Heading1
+
+        <h2>Heading2</h2></div>
+        """)
+
+        XCTAssertEqual(html, "<div><h1>Heading1</h1><h2>Heading2</h2></div>")
+    }
+    
+    func testMarkdownAfterHTML() {
+        let html = MarkdownParser().html(from: """
+        <div><h2>Heading2</h2>
+        
+        # Heading1
+        </div>
+        """)
+
+        XCTAssertEqual(html, "<div><h2>Heading2</h2><h1>Heading1</h1></div>")
+    }
+
+    
+    func testMultipleMarkdownInsideHTML() {
+        let html = MarkdownParser().html(from: """
+        <div>
+        
+        ![](image1.jpg)
+        ![](image2.jpg)
+
+        </div>
+        """)
+
+        XCTAssertEqual(html, "<div><img src=\"image1.jpg\"/><img src=\"image2.jpg\"/></div>")
+    }
+    
+    func testHTMLWithDoubleNewline() {
+        let src = """
+        <div>
+        
+        <h1>Heading</h1>
+
+        </div>
+        """
+        let html = MarkdownParser().html(from: src)
+
+        XCTAssertEqual(html, src)
+    }
+    
```

Comment:
```suggestion

    func testUnclosedHTMLWithDoubleNewline() {
        let html = MarkdownParser().html(from: """
        <div>
        *foo*

        *bar*
        """)
        XCTAssertEqual(html, "<div>\n*foo*<p><em>bar</em></p>")
    }
```

Here's an extra test that catches the out-of-bounds error.